tests/unittests/cmd/test_query.py

```diff
@@ -567,3 +567,69 @@ def test_handle_args_list_keys_errors_when_varname_is_not_a_dict(
             m_getuid.return_value = 100
             assert 1 == query.handle_args("anyname", args)
         assert expected_error in caplog.text
+
+    @pytest.mark.parametrize(
+        "header_included",
+        [ True, False ],
+    )
+    def test_handle_args_formats_jinja_successfully(
+        self, caplog, tmpdir, capsys, header_included
+    ):
+        """Test that rendering a jinja template works as expected."""
+        instance_data = tmpdir.join("instance-data")
+        instance_data.write(
+            '{"v1": {"v1_1": "val1.1", "v1_2": "val1.2"}, "v2": '
+            '{"v2_2": "val2.2"}, "top": "gun"}'
+        )
+        header = "## template: jinja\n" if header_included else ""
+        format = header + "v1_1: {{ v1.v1_1 }}"
+        expected = header + "v1_1: val1.1\n"
+        
+        args = self.Args(
+            debug=False,
+            dump_all=False,
+            format=format,
+            instance_data=instance_data.strpath,
+            list_keys=False,
+            user_data="ud",
+            vendor_data="vd",
+            varname=None,
+        )
+        with mock.patch("os.getuid") as m_getuid:
+            m_getuid.return_value = 100
+            assert 0 == query.handle_args("anyname", args)
+        out, _err = capsys.readouterr()
+        assert expected == out
+
+    def test_handle_args_invalid_jinja_exception(self, caplog, tmpdir, capsys):
+        """Raise an error when a jinja syntax error is encountered."""
+        instance_data = tmpdir.join("instance-data")
+        instance_data.write(
+            '{"v1": {"v1_1": "val1.1", "v1_2": "val1.2"}, "v2": '
+            '{"v2_2": "val2.2"}, "top": "gun"}'
+        )
+        format = "v1_1: {{ v1.v1_1 } }"
+        expected_error = (
+            "Failed to render templated data. " + 
+            JinjaSyntaxParsingException.message_template.format(
+                syntax_error="unexpected '}'", 
+                line_no=2,
+                line_content="v1_1: {{ v1.v1_1 } }"
+            )
+        )
+        args = self.Args(
+            debug=False,
+            dump_all=False,
+            format=format,
+            instance_data=instance_data.strpath,
+            list_keys=False,
+            user_data="ud",
+            vendor_data="vd",
+            varname=None,
+        )
+        with mock.patch("os.getuid") as m_getuid:
+            m_getuid.return_value = 100
+            assert 1 == query.handle_args("anyname", args)
+        out, err = capsys.readouterr()
+        assert expected_error in caplog.text
+        assert expected_error in err
```

Comment:
@a-dubs I think we determined in our discussion today that this unittest shouldn't be checking stderr setup because the query module no longer invokes `setup_basic_logging` directly to add a stderr stream handler to logging. This is now done up in [cloudinit.cmd.main before calling this submodule gets invoked](https://github.com/canonical/cloud-init/blob/main/cloudinit/cmd/main.py#L1064)

Since the logging setup is outside of this scope of this unit, I'd suggest we have a supplemental test that asserts that main does the proper setup_basic_logging on all subcommands not in `('modules', 'init')`.  We already have [separate unittest coverage that `setup_basic_logging` properly sets up a stderr handler here](https://github.com/canonical/cloud-init/blob/main/tests/unittests/test_log.py#L91). So no need to add extra stderr validation if we minimally assert when `setup_basic_logging` is called below.

```diff
--- a/tests/unittests/test_cli.py
+++ b/tests/unittests/test_cli.py
@@ -2,6 +2,7 @@
 
 import contextlib
 import io
+import logging
 import os
 from collections import namedtuple
 
@@ -172,6 +173,45 @@ class TestCLI:
         for subcommand in expected_subcommands:
             assert subcommand in err
 
+    @pytest.mark.parametrize(
+        "subcommand,log_to_stderr,mocks",
+        (
+            ("init", False, [mock.patch("cloudinit.cmd.main.status_wrapper")]),
+            (
+                "modules",
+                False,
+                [mock.patch("cloudinit.cmd.main.status_wrapper")],
+            ),
+            (
+                "schema",
+                True,
+                [
+                    mock.patch(
+                        "cloudinit.stages.Init._read_cfg", return_value={}
+                    ),
+                    mock.patch("cloudinit.config.schema.handle_schema_args"),
+                ],
+            ),
+        ),
+    )
+    @mock.patch("cloudinit.cmd.main.setup_basic_logging")
+    def test_subcommands_log_to_stderr_via_setup_basic_logging(
+        self, setup_basic_logging, subcommand, log_to_stderr, mocks
+    ):
+        """setup_basic_logging is called for modules to use stderr
+
+        Subcommands with exception of 'init'  and 'modules' use
+        setup_basic_logging to direct logged errors to stderr.
+        """
+        with contextlib.ExitStack() as mockstack:
+            for mymock in mocks:
+                mockstack.enter_context(mymock)
+            self._call_main(["cloud-init", subcommand])
+        if log_to_stderr:
+            setup_basic_logging.assert_called_once_with(logging.WARNING)
+        else:
+            setup_basic_logging.assert_not_called()
+
     @pytest.mark.parametrize("subcommand", ["init", "modules"])
     @mock.patch("cloudinit.cmd.main.status_wrapper")
     def test_modules_subcommand_parser(self, m_status_wrapper, subcommand):
```

Reply:
Patch has been applied. Thank you!